spyder/plugins/editor.py

```diff
@@ -2690,30 +2687,31 @@ def set_open_filenames(self):
             if not self.projects.get_active_project():
                 filenames = self.get_open_filenames()
                 self.set_option('filenames', filenames)
- 
+
     def setup_open_files(self):
         """Open the list of saved files per project"""
         self.set_create_new_file_if_empty(False)
         active_project_path = None
         if self.projects is not None:
-             active_project_path = self.projects.get_active_project_path()
+            active_project_path = self.projects.get_active_project_path()
 
         if active_project_path:
             filenames = self.projects.get_project_filenames()
         else:
             filenames = self.get_option('filenames', default=[])
         self.close_all_files()
- 
+
         if filenames and any([osp.isfile(f) for f in filenames]):
             self.load(filenames)
-            layout = self.get_option('layout_settings', None)
-            if layout is not None:
-                self.editorsplitter.set_layout_settings(layout)
-            win_layout = self.get_option('windows_layout_settings', None)
-            if win_layout:
+            if self.__first_open_files_setup:

```

Comment:
So this only applies the first time an Editorstack is created?

Reply:
Yes, that is the idea. Do you think there is corner cases where this can cause some problems?

Reply:
No, or at least I can't think of any right now. But that's why it'd be good to have a test for this.

Reply:
Ok, I'll see what I can do for the test.